scipy/interpolate/tests/test_predict.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import division, print_function, absolute_import
+
+import os
+import warnings
+
+import numpy as np
+from numpy.testing import assert_equal, assert_allclose, assert_almost_equal, \
+        run_module_suite, assert_raises
+
+import scipy.interpolate.predict as predict
+
+
+def test_cache_bigger_zero():
+    c = predict.Cache(0)
+    assert_raises(RuntimeError, c.add("asdf", 1))
+
+
+def test_linear_asp():
+    for length in range(20, 50, 5):
+        a = [float(item) for item in range(length)]
+        a = np.array(a)
+        res1 = predict.asp(a)
+        assert_almost_equal(res1, length)
+
+        p = predict.Cache(length)
+        for item in a:
+            p.add(item)
+        res2 = p.predict()
+        assert_almost_equal(res2, length)
+
+
+def test_square_asp():
+    for length in range(20, 50, 5):
+        a = [float(item*item) for item in range(length)]
+        a = np.array(a)
+        res1 = predict.asp(a)
+        assert res1 < length*length
+        assert 0.99*length*length < res1
+
+        p = predict.Cache(length)
+        for item in a:
+            p.add(item)
+        res2 = p.predict()
+        assert res2 < length*length
```

Comment:
Don't use plain ``assert`` statements, those will be stripped when byte-compiling with ``-O``. Use ``assert_`` from ``numpy.testing``.